bootstrap_tarball.sh

```diff
@@ -1,168 +1,204 @@
+		cp ../src/pkg/Makefile.am.pre pkg
+		cp ../src/snprintfv/bootstrap.dir ../src/snprintfv/snprintfv.m4 snprintfv
+
+		# tag the source
+		echo '5.18.987' > ZZJUNK
+		cd ../..
+	fi
+}
+
+bootstrap_columns() {
+	echo "=== Bootstrapping columns ==="
+
+	rm -Rf build/tarball
+	cp -ar build/autogen-5.18.16 build/tarball
+	cd build/tarball
+	patch -p1 -i ../../columns.patch
+	cd columns
+	mkdir -p ../../../build/stage1/bin
+	gcc columns.c -o "${PREFIX}/bin/columns"
+	cd ../../..
+}
+
+bootstrap_getdefs() {
+	echo "=== Bootstrapping getdefs ==="
+
+	rm -R build/tarball
+	cp -ar build/autogen-5.18.16 build/tarball
+	cd build/tarball
+	patch -p1 -i ../../getdefs.patch
+	cd getdefs
+	gcc -std=gnu99 getdefs.c -I ../../.. -o "${PREFIX}/bin/getdefs"
+	cd ../../..
+}
+
+bootstrap_autogen() {
+	echo "=== Bootstrapping autogen ==="
+
+	rm -R build/tarball
+	cp -ar build/autogen-5.18.16 build/tarball
+	cd build/tarball
+	patch -p1 -i ../../autogen.patch
+	cd snprintfv
+	bash bootstrap.dir
+	cd ../agen5
+	perl ../../../build-ag-text.pl
+	"${PREFIX}/bin/getdefs" output=functions.def template=functions.tpl srcfile linenum defs=macfunc listattr=alias $(grep -l '/\*=macfunc' *.c)
+	"${PREFIX}/bin/getdefs" load=expr.cfg
+	sed -n '/^doDir_invalid/d;/^doDir_[a-z]*(/{;s@(.*@@;s@^doDir_@@;p;}' defDirect.c | sort >directive_in.def
+	perl ../../../build-indirect-templates.pl
+	gcc ../../../getGuileVersion.c $(pkg-config guile-"${GUILE_VERSION}" --cflags) -o getGuileVersion
+	gcc -std=gnu99 -DGUILE_VERSION=$(./getGuileVersion) -DLIBDATADIR=\"$PREFIX/lib/autogen\" ../../../agBootstrap.c -I . -I .. -I ../../.. $(pkg-config guile-"${GUILE_VERSION}" --cflags) -o $PREFIX/bin/autogen $(pkg-config guile-"${GUILE_VERSION}" --libs)
+	cd ../../..
+}
+
+bootstrap_tpl_config() {
+	echo "=== Bootstrapping tpl-config.tlib ==="
+
+	export PATH="$PREFIX/bin:$PATH"
+
+	rm -R build/tarball
+	cp -ar build/autogen-5.18.16 build/tarball
+	cd build/tarball
+	mkdir -p ../stage1/lib/autogen
+	sed 's/@EGREP@/egrep/g;s/@GREP@/grep/g' <autoopts/tpl/tpl-config-tlib.in >"${PREFIX}/lib/autogen/tpl-config.tlib"
+	cp autoopts/tpl/*.lic "${PREFIX}/lib/autogen"
+	SOURCE_DIR="$(pwd)" ./config/bootstrap
+	./configure --prefix="$PREFIX" --disable-dependency-tracking
+	cd autoopts
+	make tpl-config-stamp
+	cp tpl/tpl-config.tlib "${PREFIX}/lib/autogen/tpl-config.tlib"
+	cd ../../..
+}
+
+rebuild_autogen() {
+	echo "=== Compiling and testing package ==="
+
+	rm -R build/tarball
+	cp -ar build/autogen-5.18.16 build/tarball
+	cd build/tarball
+	mkdir -p "${FINALPREFIX}"
+	SOURCE_DIR="$(pwd)" ./config/bootstrap
+	./configure --prefix="$FINALPREFIX" --disable-dependency-tracking
+	touch doc/agdoc.texi # build later
+	make CFLAGS=-Wno-error
+	make check
+	make install
+	cd doc
+	PATH="$FINALPREFIX/bin:$PATH"
+	rm agdoc.texi
+	make
+	make install
+	cd ../../..
+
+	echo "=== BOOTSTRAP COMPLETE ==="
+}
+
+(return 0 2>/dev/null) && sourced=1 || sourced=0
+if [ ! $sourced -eq 1 ]; then
```

Comment:
Wow, that's some convoluted hack. Is that necessary? I'd have just passed a ${SKIP_MAIN} variable from the exporting script. In any case, please uses `-ne` conditional instead of the (in my opinion) confusing `!  -eq` (or compare to zero).

Reply:
Yes, I was not particularly happy about this too... So let's use a variable